crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -590,6 +590,29 @@ pub trait StringNameSpaceImpl: AsString {
 
         Ok(substring::substring(ca, offset.i64()?, length.u64()?))
     }
+
+    /// Slice the first `n` values of the string.
+    ///
+    /// Determines a substring starting at the beginning of the string up to offset `n` of each
+    /// element in `array`. `n` can be negative, in which case the slice ends `n` characters from
+    /// the end of the string.
+    fn str_head(&self, n: &Series) -> PolarsResult<StringChunked> {
+        let ca = self.as_string();
+        let n = n.cast(&DataType::Int64)?;
```

Comment:
this should be a `strict_cast`, otherwise `UInt64` inputs might fail silently.

Reply:
I've updated the cast in `head` and `tail` in the meanwhile.

I think I can do a minor improvement to the `slice` algorithm if we want to put those in a separate PR. I'll need someone with more knowledge to chime in as to whether it'll actually be faster though (integer comparisons as in the `head`/`tail` in this PR versus using `iter_chars.nth` as `slice` does).